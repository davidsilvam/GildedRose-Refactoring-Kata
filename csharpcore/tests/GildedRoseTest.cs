using Xunit;
using System.Collections.Generic;
using Xunit.Abstractions;//added to debug

using ApprovalTests;
using ApprovalTests.Reporters;

using System.Text;

namespace csharpcore
{
    [UseReporter(typeof(DiffReporter))]
    public class GildedRoseTest
    {
        private readonly ITestOutputHelper _output;

        public GildedRoseTest(ITestOutputHelper output)
        {
            _output = output;
        }

        [Fact]
        public void foo()
        {
            IList<Item> Items = new List<Item> { new Item { Name = "foo", SellIn = 0, Quality = 0 } };
            GildedRose app = new GildedRose(Items);
            app.UpdateQuality();
            //  _output.WriteLine("Debug");
            Assert.Equal("foo", Items[0].Name);
        }

        [Fact]
        public void NormalItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
                new Item {
                    Name = "+5 Dexterity Vest",
                    SellIn = 10,
                    Quality = 20
                },
                new Item {
                    Name = "Elixir of the Mongoose",
                    SellIn = 5,
                    Quality = 7
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }

        [Fact]
        public void AgedBrieItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
                new Item {
                    Name = "Aged Brie",
                    SellIn = 2,
                    Quality = 0
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }

        [Fact]
        public void SulfurasItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
                new Item {
                    Name = "Sulfuras, Hand of Ragnaros",
                    SellIn = 0,
                    Quality = 80
                },
                new Item {
                    Name = "Sulfuras, Hand of Ragnaros",
                    SellIn = -1,
                    Quality = 80
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }

        [Fact]
        public void BackstagePassesItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
               new Item
                {
                    Name = "Backstage passes to a TAFKAL80ETC concert",
                    SellIn = 15,
                    Quality = 20
                },
                new Item
                {
                    Name = "Backstage passes to a TAFKAL80ETC concert",
                    SellIn = 10,
                    Quality = 49
                },
                new Item
                {
                    Name = "Backstage passes to a TAFKAL80ETC concert",
                    SellIn = 5,
                    Quality = 49
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }

        [Fact]
        public void ConjuredItemIncreaseAndDecrease()
        {
            IList<Item> Items = new List<Item> {
               new Item {
                   Name = "Conjured Mana Cake",
                   SellIn = 3,
                   Quality = 6
                }
            };

            GildedRose app = new GildedRose(Items);

            var fakeoutput = new StringBuilder();

            for (var i = 0; i < 31; i++)
            {
                fakeoutput.Append("-------- day " + i + " --------" + "\n");
                for (var j = 0; j < Items.Count; j++)
                {
                    fakeoutput.Append(Items[j].Name + ", " + Items[j].SellIn + ", " + Items[j].Quality + "\n");
                }
                app.UpdateQuality();
            }

            Approvals.Verify(fakeoutput);
        }
    }
}