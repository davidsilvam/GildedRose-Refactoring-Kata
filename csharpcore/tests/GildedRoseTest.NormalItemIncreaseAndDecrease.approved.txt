-------- day 0 --------
+5 Dexterity Vest, 10, 20
Elixir of the Mongoose, 5, 7
-------- day 1 --------
+5 Dexterity Vest, 9, 19
Elixir of the Mongoose, 4, 6
-------- day 2 --------
+5 Dexterity Vest, 8, 18
Elixir of the Mongoose, 3, 5
-------- day 3 --------
+5 Dexterity Vest, 7, 17
Elixir of the Mongoose, 2, 4
-------- day 4 --------
+5 Dexterity Vest, 6, 16
Elixir of the Mongoose, 1, 3
-------- day 5 --------
+5 Dexterity Vest, 5, 15
Elixir of the Mongoose, 0, 2
-------- day 6 --------
+5 Dexterity Vest, 4, 14
Elixir of the Mongoose, -1, 0
-------- day 7 --------
+5 Dexterity Vest, 3, 13
Elixir of the Mongoose, -2, 0
-------- day 8 --------
+5 Dexterity Vest, 2, 12
Elixir of the Mongoose, -3, 0
-------- day 9 --------
+5 Dexterity Vest, 1, 11
Elixir of the Mongoose, -4, 0
-------- day 10 --------
+5 Dexterity Vest, 0, 10
Elixir of the Mongoose, -5, 0
-------- day 11 --------
+5 Dexterity Vest, -1, 8
Elixir of the Mongoose, -6, 0
-------- day 12 --------
+5 Dexterity Vest, -2, 6
Elixir of the Mongoose, -7, 0
-------- day 13 --------
+5 Dexterity Vest, -3, 4
Elixir of the Mongoose, -8, 0
-------- day 14 --------
+5 Dexterity Vest, -4, 2
Elixir of the Mongoose, -9, 0
-------- day 15 --------
+5 Dexterity Vest, -5, 0
Elixir of the Mongoose, -10, 0
-------- day 16 --------
+5 Dexterity Vest, -6, 0
Elixir of the Mongoose, -11, 0
-------- day 17 --------
+5 Dexterity Vest, -7, 0
Elixir of the Mongoose, -12, 0
-------- day 18 --------
+5 Dexterity Vest, -8, 0
Elixir of the Mongoose, -13, 0
-------- day 19 --------
+5 Dexterity Vest, -9, 0
Elixir of the Mongoose, -14, 0
-------- day 20 --------
+5 Dexterity Vest, -10, 0
Elixir of the Mongoose, -15, 0
-------- day 21 --------
+5 Dexterity Vest, -11, 0
Elixir of the Mongoose, -16, 0
-------- day 22 --------
+5 Dexterity Vest, -12, 0
Elixir of the Mongoose, -17, 0
-------- day 23 --------
+5 Dexterity Vest, -13, 0
Elixir of the Mongoose, -18, 0
-------- day 24 --------
+5 Dexterity Vest, -14, 0
Elixir of the Mongoose, -19, 0
-------- day 25 --------
+5 Dexterity Vest, -15, 0
Elixir of the Mongoose, -20, 0
-------- day 26 --------
+5 Dexterity Vest, -16, 0
Elixir of the Mongoose, -21, 0
-------- day 27 --------
+5 Dexterity Vest, -17, 0
Elixir of the Mongoose, -22, 0
-------- day 28 --------
+5 Dexterity Vest, -18, 0
Elixir of the Mongoose, -23, 0
-------- day 29 --------
+5 Dexterity Vest, -19, 0
Elixir of the Mongoose, -24, 0
-------- day 30 --------
+5 Dexterity Vest, -20, 0
Elixir of the Mongoose, -25, 0
